src/main/java/io/github/binaryoverload/JSONConfig.java

```diff
@@ -118,17 +117,18 @@ public JSONConfig(InputStream stream) {
      *                      null, empty or any other lenth than 1</i>
      * @throws NullPointerException     if any of the passed arguments are null
      * @throws IllegalArgumentException if the path separator is empty or not a length of 1
-     * @throws IOException              if the stream is invalid or malformatted 
+     * @throws IOException              if the stream is invalid or malformatted
      * @see InputStream
-     * @see org.apache.commons.io.IOUtil
      * * @since 1.0
      */
     public JSONConfig(InputStream stream, String pathSeparator) throws IOException {
         Objects.requireNonNull(stream);
         Objects.requireNonNull(pathSeparator);
         GeneralUtils.checkStringLength(pathSeparator, 1);
         setPathSeparator(pathSeparator);
-        this.object = GSON.fromJson(IOUtil.toString(stream), JsonObject.class);
+        BufferedReader br = new BufferedReader(new InputStreamReader(stream));
+        this.object = GSON.fromJson(br.lines().collect(Collectors.joining("\n")), JsonObject.class);
```

Comment:
We could technically just join the strings, it would essentially be semi-minified JSON

Reply:
True